Assets/UI/WorldView/citybannermanager_CQUI_basegame.lua

```diff
@@ -246,32 +230,40 @@ function CityBanner.UpdateName( self )
-                    if (districtType == iCommerce)        then self.m_Instance.CityBuiltDistrictCommercial:SetHide(false);     end
-                    if (districtType == iEncampment)      then self.m_Instance.CityBuiltDistrictEncampment:SetHide(false);     end
-                    if (districtType == iEntComplex)      then self.m_Instance.CityBuiltDistrictEntertainment:SetHide(false);  end
-                    if (districtType == iHansa)           then self.m_Instance.CityBuiltDistrictHansa:SetHide(false);          end
-                    if (districtType == iHarbor)          then self.m_Instance.CityBuiltDistrictHarbor:SetHide(false);         end
-                    if (districtType == iHolySite)        then self.m_Instance.CityBuiltDistrictHoly:SetHide(false);           end
-                    if (districtType == iIndustrial)      then self.m_Instance.CityBuiltDistrictIndustrial:SetHide(false);     end
-                    if (districtType == iLavra)           then self.m_Instance.CityBuiltDistrictLavra:SetHide(false);          end        
-                    if (districtType == iMbanza)          then self.m_Instance.CityBuiltDistrictMbanza:SetHide(false);         end
-                    if (districtType == iNeighborhood)    then self.m_Instance.CityBuiltDistrictNeighborhood:SetHide(false);   end
-                    if (districtType == iRoyalNavy)       then self.m_Instance.CityBuiltDistrictRoyalNavy:SetHide(false);      end
-                    if (districtType == iSpaceport)       then self.m_Instance.CityBuiltDistrictSpaceport:SetHide(false);      end
-                    if (districtType == iStreetCarnival)  then self.m_Instance.CityBuiltDistrictStreetCarnival:SetHide(false); end
-                    if (districtType == iTheater)         then self.m_Instance.CityBuiltDistrictTheater:SetHide(false);        end
+                    -- Items in the districts table are populated above; it does not include City Center or 
+                    if (districts[districtType] ~= nil
+                       and (districts[districtType].Instance ~= self.m_Instance.CityBuiltDistrictNeighborhood or neighborhoodAdded == false)) then
+                        districtTooltipString = districtTooltipString .. districts[districtType].Icon .. " " .. Locale.Lookup(districtInfo.Name) .. "[NEWLINE]";
+                        districtsBuilt = districtsBuilt + 1;
+                        districts[districtType].Instance:SetHide(false);
+                        if (districts[districtType].Instance == self.m_Instance.CityBuiltDistrictNeighborhood) then
+                            neighborhoodAdded = true;
+                        end
+                    end
                 end -- if isBuilt
             end -- for loop
+
+            -- Trim the trailing [NEWLINE]
+            districtTooltipString = string.sub(districtTooltipString, 1, string.len(districtTooltipString) - 9);
```

Comment:
In case anyone else was wondering: this produces an empty string in the situation where districtTooltipString is still empty by this point. This should be fine for our purposes.